src/Server/BaseServer.ts

```diff
@@ -141,15 +142,66 @@ export class BaseServer {
    * Script object on the server (if it exists)
    */
   getScript(scriptName: string): Script | null {
+    const filename = sanitizeFilename(scriptName);
     for (let i = 0; i < this.scripts.length; i++) {
-      if (this.scripts[i].filename === scriptName) {
+      if (this.scripts[i].filename === filename) {
         return this.scripts[i];
       }
     }
 
     return null;
   }
 
+  /**
+   * @returns Array of all script filenames on server
+   */
+  getAllScriptFilenames(): string[] {
+    const filenames: string[] = [];
+    for (const file of this.scripts) {
+      filenames.push(file.filename);
+    }
+    return filenames;
+  }
+
+  /**
+   * @returns Array of all script files on server
+   */
+  getAllScriptFiles(): Script[] {
+    return [...this.scripts];
+  }
+
+  /**
+   * Given the name of the textfile, returns the corresponding
+   * TextFile object on the server (if it exists)
+   */
+  getFile(textFileName: string): TextFile | null {
+    const filename = sanitizeFilename(textFileName);
+    for (let i = 0; i < this.textFiles.length; i++) {
+      if (this.textFiles[i].filename === filename) {
+        return this.textFiles[i];
+      }
+    }
+    return null;
+  }
+
+  /**
+   * @returns Array of all textfile filenames on server
+   */
+  getAllTextFilenames(): string[] {
+    const filenames: string[] = [];
+    for (const file of this.textFiles) {
```

Comment:
nit: maybe a map here too

Reply:
propbably, theres a lot of places in the codebase where a plain object or array is used when a Map would make a lot of sense

Reply:
yeah just figured might aswell fix the ones that your touching to keep your pr as small as possible its already huge 👍